kvmagent/kvmagent/plugins/vm_plugin.py

```
@@ -10819,6 +10829,120 @@ def find_mdev_device(vm_uuid, mdev_uuid):
+            if self._is_dgpu_shmem_attached(vm, shmem_name):
+                logger.debug("dGPU shmem[%s] already attached to vm[uuid:%s]" % (shmem_name, cmd.vmUuid))
+                return jsonobject.dumps(rsp)
+
+            vm.domain.attachDeviceFlags(xml, libvirt.VIR_DOMAIN_AFFECT_LIVE)
+            if not self._wait_dgpu_shmem_attached_state(vm, shmem_name, True):
+                rsp.success = False
+                rsp.error = "dGPU shmem %s still not attached to vm %s after 20s" % (shmem_name, cmd.vmUuid)
+                return jsonobject.dumps(rsp)
+
+            logger.debug("attached dGPU shmem[%s] to vm[uuid:%s]" % (shmem_name, cmd.vmUuid))
+        except Exception as ex:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)